src/ORM/Association.php

```diff
@@ -391,7 +391,7 @@ public function getTarget(): Table
             $config = [];
             $exists = $tableLocator->exists($registryAlias);
             if (!$exists) {
-                $config = ['className' => $this->_className];
+                $config = ['className' => $this->_className, 'connection' => $this->getSource()->getConnection()];
```

Comment:
Shouldn't the concrete class for the association target define its connection?

Reply:
Explicitly setting the connection here would mean it would override any non default connection specified in the table class isn't it? That isn't good.

Reply:
What do you propose here? Any associated generated from a key would use the same connection.

Reply:
Maybe check after `get()` if `$this->_targetTable` is `TableLocator::$fallbackClassName` (needs a public getter), and if so, set the connection on the constructed table object?

Reply:
The issue I'm trying to solve is the bake issue noted in the comment. We override the connection which is used everywhere for Table creation except here where the table doesn't exist before calling `getTarget()`.

In those cases, the default connection is overridden by the `-c` connection option.

Reply:
Ah, sorry, I should've read the initial post. If my concrete table class should use the default connection, then I certainly wouldn't want that to change in my app.

Maybe this is something that `ModelCommand::getTableObject()` should enforce? Currently it's only doing that for not yet constructed table objects, meaning if they get constructed somewhere else in the code before that, they will have the "wrong" connection, assuming the user actually expects `-c` to modify the connections of all involved tables.

Reply:
Are you suggesting that getTableObject() walk through all associations and set the connection flag? I'm not sure that's the order of operations in ModelCommand which is generating the associations.

Reply:
No, I mean just the table that `getTableObject()` obtains. Not sure if I'm missing something, but aren't the tables obtained via that method the only ones that the command actually makes use of with regards to establishing connections?